beacon-chain/blockchain/process_block.go

```diff
@@ -298,3 +313,113 @@ func (s *Service) insertBlockToForkChoiceStore(ctx context.Context, blk *ethpb.B
 
 	return nil
 }
+
+func (s *Service) persistCachedStates(ctx context.Context, numOfStates int) error {
```

Comment:
function comment please, and maybe a better place for this and below are in `process_block_helpers.go` ?

Reply:
or even better `init_sync_process_block.go` or something more initial sync related. If this is only used by initial sync

Reply:
moved all functions to a separate file